### PR TITLE
Added a timeout for the sponsor to click on the link

### PR DESCRIPTION
**Pull request by fdurand**
# Description

Added an activation timeout on sponsor source
# Impacts

Sponsor source
# Delete branch after merge

YES
# NEWS file entries
## Enhancements
- Added an activation timeout on sponsor source

### CODE REVIEW

File: html/pfappserver/lib/pfappserver/Form/Config/Authentication/Source/SponsorEmail.pm [hunk 1]
```diff
@@ -28,6 +28,16 @@ has_field 'allow_localdomain' =>
...
   );
 
+has_field 'email_activation_timeout' =>
+  (
+   type => 'Duration',
+   label => 'Email Activation Timeout',
+   required => 1,
+   default => pfappserver::Form::Field::Duration->duration_inflate(pf::Authentication::Source::EmailSource->meta->get_attribute('email_activation_timeout')->default),
+   tags => { after_element => \&help,
+             help => 'This is the delay given to a sponsor to click the activation link.' },
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/Form/Config/Authentication/Source/SponsorEmail.pm]: Change to 'Delay given to a sponsor to click the activation link.'

File: html/pfappserver/lib/pfappserver/Form/Config/Authentication/Source/SponsorEmail.pm [hunk 2]
```diff
@@ -28,6 +28,16 @@ has_field 'allow_localdomain' =>
              help => 'Accept self-registration with email address from the local domain' },
   );
 
+has_field 'email_activation_timeout' =>
+  (
+   type => 'Duration',
+   label => 'Email Activation Timeout',
+   required => 1,
+   default => pfappserver::Form::Field::Duration->duration_inflate(pf::Authentication::Source::EmailSource->meta->get_attribute('email_activation_timeout')->default),
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/Form/Config/Authentication/Source/SponsorEmail.pm]: Should be taken from SponsorSource not EmailSource

File: html/pfappserver/lib/pfappserver/Form/Config/Authentication/Source/SponsorEmail.pm [hunk 3]
```diff
@@ -28,6 +28,16 @@ has_field 'allow_localdomain' =>
              help => 'Accept self-registration with email address from the local domain' },
   );
 
+has_field 'email_activation_timeout' =>
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/Form/Config/Authentication/Source/SponsorEmail.pm:31]: You need to add the field `email_activation_timeout` to the SponsorEmailSource module as it doesn't seem to be there..

**Review by julsemaan — CHANGES_REQUESTED** [review]

**Comment by fdurand** [comment]
Done

**Pull request merged**